### PR TITLE
Custom movement hotkeys and restore macros for T, O and M

### LINKED ISSUE #47621: some issues with custom hotkeys
1. the ooc/say boxes dont have a cancel/X button, pressing escape doesnt close out of them either
2. maybe unrelated to custom hotkeys but definitely wasnt a thing in the past: playing on a server with 200 ping, i notice a considerable delay from pressing T or O until i'm able to type. in the past, i would be able to type instantly, regardless of ping, this is annoying for obvious reasons. i play on servers with this ping frequently, and have not noticed it at all before.
3. pressing M just says "unusable emote: me" in chat.

### PR DESCRIPTION
**Pull request by Couls**
## About The Pull Request
Restores the T, O and M macros so that you can escape out of chat windows and they have an X in the corner, this also prevents the latency you get when you have a high ping and attempt to use T, O or M hotkeys.The downside is that these three keys are no longer customizable. Also adds in custom movement hotkeys
should fix #47621 

## Why It's Good For The Game
Not every keyboard is QWERTY and this allows people to change the movement keys to whatever they feel comfortable with

## Changelog
:cl:
add: Movement keys are now customizable
del: T, O and M are no longer customizable
del: admin commands for messing with movement keys, this won't work since they're different for each client
fix: ooc, say and me are all escapable now and no longer lag at higher pings
/:cl:

### CODE REVIEW

File: code/datums/keybinding/movement.dm [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/datum/keybinding/movement
+    category = CATEGORY_MOVEMENT
+    weight = WEIGHT_HIGHEST
+
+/datum/keybinding/movement/north
+	key = "W"
+	name = "North"
+	full_name = "Move Up"
+	description = "Moves your character north"
```
Comment (MrPerson) [inline code/datums/keybinding/movement.dm]: The name says "up" and the description says "north". LITERALLY UNPLAYABLE

Comment (Couls) [reply code/datums/keybinding/movement.dm]: If you don't know what north means you'll definitely know what up is

File: code/controllers/subsystem/input.dm [hunk 2]
```diff
@@ -25,24 +22,20 @@ SUBSYSTEM_DEF(input)
...
-
-	movement_keys = default_movement_keys.Copy()
-
 // Badmins just wanna have fun ♪
 /datum/controller/subsystem/input/proc/refresh_client_macro_sets()
 	var/list/clients = GLOB.clients
 	for(var/i in 1 to clients.len)
 		var/client/user = clients[i]
 		user.set_macros()
+		user.update_movement_keys()
```
Comment (Couls) [inline code/controllers/subsystem/input.dm]: don't know if I should rename the proc now that it also updates movement keys as it sets macros

**Comment by Tharcoonvagh** [comment]
Thank god I can play again

**Comment by actioninja** [comment]
You could still make these rebindable a different way, I've got an idea once this is working and merged.

**Comment by Couls** [comment]
Looking forward to it I’m interested to see how you’ll do it

**Comment by Couls** [comment]
Should be ready for review, I kind of want ninja to take a look at it because he's done stuff with input before and this code touches some of the code in the input subsystem

**Review by actioninja — CHANGES_REQUESTED** [review]

**Review by actioninja — APPROVED** [review]

**Comment by Takov-Mala** [comment]
You borked everything. Everything is actually unbinded now and people have to change hotkeys in preferences to even move. 
Also add a way to just blank some hotkeys, a bunch of these I don't even want.

**Comment by Couls** [comment]
There is a way to unbind keys it explains how when you click on the key you wish to unbind

**Comment by Takov-Mala** [comment]
Ah, cool.

**Comment by peoplearestrange** [comment]
Bullshit

**Comment by Couls** [comment]
what?

**Pull request merged**